Automattic/jetpack
# Add all contact form values to “to” field hook

Fixes #25403

#### Changes proposed in this Pull Request:

Add $all_values param to contact_form_to filter


#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?
No.




#### Testing instructions:

In order to test this pr, you need to be on "Online" mode, as the "Contact Form" block is enabled only when being online.
The change we are making is to improve code extendability, so we propose testing it using a small snippet.







- Run jetpack locally
- Add a `Contact form` block to your page.
- Navigate to `projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php` file in your ide.
- Before the code that was added add the following code snippet:

  ```
    add_filter(
      'contact_form_to',
      function ( $to, $all_values) {
		    error_log(var_export($all_values, true));
		    return [];
      }, 10, 2
    );
  ```

- Visit the page you have created and submit the form.
- Run `jetpack docker tail` and make sure you see the printed `$all_values` variable. It should
  look something like the following:

  ```
  array (
    '1_Name' => 'test',
    '2_Email' => 'test@test.com',
    '3_Message' => 'tse',
  )
  ```
